### PR TITLE
[#261] Added API integration, removed demands from similarity

### PR DESCRIPTION
**Pull request by arekmano**
- Added ML service API integration for similiarity finder
- Removed demands from similarity finder, as we are never supplied with it
- Changed the distance metric from mahalanobi to euclidean with a scaler, as mahalanobi requires a covariance matrix, which we cannot calculate with our dataset (memory requirements are too high)

### CODE REVIEW

File: src/ml_service/model_training/similar_finder/similar_finder.py [hunk 1]
```diff
@@ -17,19 +18,23 @@ def __init__(self, train=False, dataset=[]):
...
+            sample_set = [np.concatenate(
+                [vec['facts_vector'], vec['outcomes_vector']]) for vec in dataset]
 
             for i in range(len(sample_set)):
                 sample_set[i] = sample_set[i].astype(np.int64)
 
-            self.model = NearestNeighbors(5, metric='mahalanobis', metric_params={'V': np.cov(sample_set)})
+            self.scaler = StandardScaler()
+            sample_set = self.scaler.fit_transform(sample_set)
+            self.model = NearestNeighbors(5, metric='euclidean')
```
Comment (Samuel-Campbell) [inline src/ml_service/model_training/similar_finder/similar_finder.py:31]: scaler + euclidean distance makes sense

File: src/ml_service/web/ml_controller.py [hunk 2]
```diff
@@ -31,8 +33,10 @@ def predict_outcome(input_json):
         """
         facts_vector = MlController.dict_to_vector(input_json['facts'])
         outcome_vector = GlobalPredictor.predict_outcome(facts_vector)
-
-        return MlController.vector_to_dict(outcome_vector)
+        response = MlController.vector_to_dict(outcome_vector)
+        similar_dict = {'facts_vector': facts_vector, 'outcomes_vector': outcome_vector}
```
Comment (Samuel-Campbell) [inline src/ml_service/web/ml_controller.py:37]: I think we should make a different POST request for this. We are not predicting an outcome here

Comment (arekmano) [reply src/ml_service/web/ml_controller.py:37]: I included it in the /predict endpoint since the NLP service will only be calling this at the end, to obtain the overall results page. SimilarFinder uses the same inputs and will be shown at the same time as the predictions, so it makes sense to include them in the same endpoint.

Comment (Samuel-Campbell) [reply src/ml_service/web/ml_controller.py:37]: ohhh right got'cha. It's true that the inputs are the same

**Review by Samuel-Campbell — APPROVED** [review]

File: src/ml_service/model_training/similar_finder/similar_finder.py [hunk 3]
```diff
@@ -40,7 +45,7 @@ def __init__(self, train=False, dataset=[]):
             e.g. {
```
Comment (TaimoorRana) [inline src/ml_service/model_training/similar_finder/similar_finder.py:45]: Need to move the comments underneath the method header as we are doing this for all the other files. Not a blocker

**Review by TaimoorRana — APPROVED** [review]
Looks good. Just have a look at my comment
